replay/models/als.py

```diff
@@ -28,13 +28,16 @@ def _get_ann_infer_params(self) -> Dict[str, Any]:
             "index_dim": self.rank,
         }
 
-    def _get_vectors_to_infer_ann(self, log: DataFrame, users: DataFrame) -> DataFrame:
+    def _get_vectors_to_infer_ann(
+        self, log: DataFrame, users: DataFrame, filter_seen_items: bool
+    ) -> DataFrame:
         user_vectors, _ = self.get_features(users)
-        user_to_max_items = (
-            log.groupBy('user_idx')
-            .agg(sf.count('item_idx').alias('num_items'))
-        )
-        user_vectors = user_vectors.join(user_to_max_items, on="user_idx")
+        if filter_seen_items:
```

Comment:
this part is duplicated in ALS, worrd2vec, knn, SLIM.
Let's move it to the base ANNMixin by doing refactoring the following way:
- introduce _get_vectors_to_infer_ann_inner, that will play a role of what _get_vectors_to_infer_ann does now
-  add a new impl of _get_vectors_to_infer_ann that will consists of calling '_get_vectors_to_infer_ann_inner' to obtain user_vectors and than handling of fitlter_seen_items flag